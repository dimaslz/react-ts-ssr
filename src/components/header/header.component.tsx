import { Link } from "@/components";

export const Header = () => (
	<header
		className={`fixed left-0 top-0 z-50 flex h-16 w-full items-center bg-gray-800 p-0 text-white shadow-sm dark:bg-gray-900`}
	>
		<h1 className="relative flex w-full justify-start pl-6">Logo</h1>
		<nav className="flex h-full items-center text-sm">
			<Link
				withBackground
				href="/"
				className="flex h-full items-center justify-center px-4 py-0"
			>
				Home
			</Link>
			<Link
				withBackground
				href="/other"
				className="flex h-full items-center justify-center px-4 py-0"
			>
				Other
			</Link>
			<Link
				withBackground
				href="/projects"
				className="flex h-full items-center justify-center px-4 py-0"
			>
				Projects
			</Link>
		</nav>
	</header>
);
