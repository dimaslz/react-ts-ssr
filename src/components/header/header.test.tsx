import { render } from "@testing-library/react";

import { Header } from "./header.component";

describe("Header", () => {
	test("snapshot", () => {
		const component = render(<Header />);

		expect(component).toMatchSnapshot();
	});

	test("has three items", () => {
		const component = render(<Header />);

		expect(component.getAllByRole("link")).toHaveLength(3);
	});
});
