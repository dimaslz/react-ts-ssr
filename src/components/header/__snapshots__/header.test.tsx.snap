// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="fixed left-0 top-0 z-50 flex h-16 w-full items-center bg-gray-800 p-0 text-white shadow-sm dark:bg-gray-900"
      >
        <h1
          class="relative flex w-full justify-start pl-6"
        >
          Logo
        </h1>
        <nav
          class="flex h-full items-center text-sm"
        >
          <a
            class="flex h-full items-center justify-center px-4 py-0 hover:bg-gray-900 dark:hover:bg-gray-700"
            href="/"
          >
            Home
          </a>
          <a
            class="flex h-full items-center justify-center px-4 py-0 hover:bg-gray-900 dark:hover:bg-gray-700"
            href="/other"
          >
            Other
          </a>
          <a
            class="flex h-full items-center justify-center px-4 py-0 hover:bg-gray-900 dark:hover:bg-gray-700"
            href="/projects"
          >
            Projects
          </a>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="fixed left-0 top-0 z-50 flex h-16 w-full items-center bg-gray-800 p-0 text-white shadow-sm dark:bg-gray-900"
    >
      <h1
        class="relative flex w-full justify-start pl-6"
      >
        Logo
      </h1>
      <nav
        class="flex h-full items-center text-sm"
      >
        <a
          class="flex h-full items-center justify-center px-4 py-0 hover:bg-gray-900 dark:hover:bg-gray-700"
          href="/"
        >
          Home
        </a>
        <a
          class="flex h-full items-center justify-center px-4 py-0 hover:bg-gray-900 dark:hover:bg-gray-700"
          href="/other"
        >
          Other
        </a>
        <a
          class="flex h-full items-center justify-center px-4 py-0 hover:bg-gray-900 dark:hover:bg-gray-700"
          href="/projects"
        >
          Projects
        </a>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
